samza-core/src/main/java/org/apache/samza/coordinator/MetadataResourceManager.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.coordinator;
+
+import org.apache.samza.checkpoint.CheckpointManager;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.TaskConfigJava;
+import org.apache.samza.coordinator.metadatastore.NamespaceAwareCoordinatorStreamStore;
+import org.apache.samza.coordinator.stream.messages.SetChangelogMapping;
+import org.apache.samza.job.model.JobModel;
+import org.apache.samza.metadatastore.MetadataStore;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.startpoint.StartpointManager;
+import org.apache.samza.storage.ChangelogStreamManager;
+
+
+/**
+ * Loads the managers responsible for the creation and loading of metadata related resources.
+ * TODO: Replace with a metadata admin interface when the {@link MetadataStore} is fully augmented to handle all metadata sources.
+ */
+public class MetadataResourceManager {
```

Comment:
Is this concept specific to Kafka streams? E.g., what does it mean to create resources if the metadata store is a DB? If the former, can we do this as a util instead of a new interface?

Reply:
No, it's to consolidate managers that interact with the metadata store or other metadata streams. It is essentially a util class. I can change the name to `*Util` instead of `*Manager` if that makes better sense. Ideally, when the metadata store extensions are fully designed and implemented, that should replace this.

Reply:
What about LocalityManager / TaskAssignmentManager etc? Should they be managed by this class as well? Looking at just this class in isolation, it's unclear what its responsibilities are. +1 for calling it XUtil if it's temporary and for code cleanup/consolidation.

Reply:
Any suggestions to make the responsibility more obvious? I already renamed the class to `*Util` and already have a `TODO:` comment in the document to replace this in the future once the full design and implementation of the metadata store is in place. Think it's sufficient for now?

Reply:
My bad, was looking at outdated version. Sounds good to me, thanks.